### PR TITLE
Refactors `ledger` and `synthesizer` crates into smaller dependencies

### PR DESCRIPTION
**Pull request by howardwu**
## Motivation

This PR refactors `ledger` and `synthesizer` crates into smaller dependencies. There is notable improvements in compile time performance, as well as CI runtime performance.

In the process several unnecessary dependencies were removed, further improving performance.

### Changes

One notable change was introduced in the process for the `Call` instruction. Due to compiler requirements, the `Call::evaluate` and `Call::execute` implementations have been traitified and separated from its primary implementation. These 2 methods can be found in the `synthesizer/process/src/stack/call/mod.rs`.

The following is a summary of the migrations:
- `synthesizer/src/process` -> `synthesizer/process`
- `synthesizer/src/block` -> `ledger/block`
- `synthesizer/src/query` -> `ledger/query`
- `synthesizer/src/store` -> `ledger/store`

### Testing

A new `ledger-test-helpers` crate was introduced in the `ledger/` folder to accommodate the new refactor into smaller dependencies. As part of this, many prior unit tests have been updated to use the test helper methods defined in the new `ledger-test-helpers` crate.

### Next Steps

There remains one final remaining refactor that is still necessary. In a follow-on PR, the `synthesizer/coinbase` crate will be moved to `ledger/coinbase` to facilitate the correct location for this logic (as it is related to `block`).

### CODE REVIEW

File: synthesizer/src/vm/execute.rs
Comment (raychu86) [inline synthesizer/src/vm/execute.rs:1]: Nit: Not part of this PR, but documentation for `VM::execute` should be updated - replace "deployment" with "execution".

> /// The `priority_fee_in_microcredits` is an additional fee **on top** of the ~~deployment~~ execution fee.

**Review by raychu86 — APPROVED** [review]
LGTM

File: ledger/query/src/traits.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use console::{network::Network, prelude::Result, program::StatePath, types::Field};
+
+#[cfg_attr(feature = "async", async_trait)]
+pub trait QueryTrait<N: Network> {
```
Comment (iamalwaysuncomfortable) [inline ledger/query/src/traits.rs:18]: This type of "provider" trait is something that might be applicable in a few key places elsewhere in SnarkVM as well

**Review by iamalwaysuncomfortable — APPROVED** [review]
